openpype/hosts/fusion/plugins/create/create_saver.py

```diff
@@ -119,10 +119,9 @@ def _update_tool_with_data(self, tool, data):
         if "subset" not in data:
             return
 
-        original_subset = tool.GetData("openpype.subset")
-        subset = data["subset"]
-        if original_subset != subset:
-            self._configure_saver_tool(data, tool, subset)
+        original_data = tool.GetData("openpype")
+        if original_data != data["creator_attributes"]:
+            self._configure_saver_tool(data, tool, data["subset"])
```

Comment:
This if statement will always be true since `openpype` data stored is more than `creator_attributes` alone.
So in this case you're ALWAYS forcing again the creator settings - even if user changed nothing which can be unexpected.

I suspect there's a logic error here and you intended different. If you intended it like this, then the if statement isn't needed to begin with.

Reply:
I think we're better of doing this:

Split the `_configure_saver_tool` into:
- _update_subset_change()
- _update_file_format_change()

Then I suppose here we would do:

```python
if file_format_changed:  # basically: ext != current_ext
    # Updating extension
    self._update_file_format_change()

if subset_changed:  # basically: subset != current_subset
    # Updating filepath (keeping whatever extension it had from above logic?)
    # and renaming the tool to match subset
    self.update_subset_change()
```

Reply:
Correct, it was a typo on my end (forgot to take lunch so was rushing about...). I'm kind of split as the original statement checking if subset changed is made redundant by the comparison of creator_attributes changing so really the statement would only exist for clarity.

Reply:
I think it might make sense to break up `_configure_saver_tool` into multiple functions and then as you say have some different methods for updating the saver. That way it's not always updating everything. Are we happy with this general approach though, does it match the feel of other DCC's in the way it feels for users?

Reply:
Likely we'd need to compare with Houdini or Nuke (hosts which specify export paths in their scenes). @MustafaJafar @jakubjezek001 any views from that angle?